### ISSUE TITLE
Linting error on LT01 and LT02 appears in the CI but not in local enviroment

### ISSUE THREAD

**Issue by nicor88** [sqlfluff/sqlfluff#5098]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

My CI (in bitbucket pipelines) show some linting errors that are not shown locally (mac os ubuntu).

Specifically the errors seems only related to: LT02 and LT01 - that is super odd.
Somehow this seems related: https://github.com/sqlfluff/sqlfluff/issues/1847#issuecomment-1432722891

sqlfluff version

### Expected Behaviour

I expect that the linter is environment agnostic

### Observed Behaviour

this behavior don't happen when working with jijia templater, but only when using dbt templater - needed to work with external macros properly.

### How to reproduce

not sure how you can reproduce, but I expect that at least error LT01 is catched- somehow the rendering of the model seems not environment agnostic

### Dialect

athena

### Version

2.3.0 and dbt templater

### Configuration

[sqlfluff]
dialect = athena
exclude_rules = L034,RF02,ST06
max_line_length = 120
templater = dbt

[sqlfluff:indentation]
tab_space_size = 4
indent_unit = space

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = trailing

[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

[sqlfluff:templater:dbt]
project_dir = ./

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by svdimchenko** [comment]
I've noticed that it fails when following condition is added:
```
    where
       wallet_id is not null
    {%- if is_incremental() %}
        and etl_ts = {{ get_etl_ts() }}
    {%- endif %}
```

when actual table exists, I get the error:
```
L:  24 | P:   1 | LT02 | Expected indent of 4 spaces.
```

however when table does not exist, sqlfluff check passes without errors


And fix is following:
```
    where
        wallet_id is not null
    {%- if is_incremental() %}
    and etl_ts = {{ get_etl_ts() }}
    {%- endif %}
```
which looks like a bit strange

**Comment by alanmcruickshank** [comment]
I think this is very related to https://github.com/sqlfluff/sqlfluff/issues/4061 which I'm in the process of trying to get live. Won't happen immediately, but I think we have a solution in the works.

**Comment by francois-blanchard** [comment]
Hey 👋 we have the same issue.
Any update about it?

Here is our example:
```
where
    date_hour_at::date <= expiry_date::date + 3
    and date_hour_at <= current_timestamp - interval '150 minutes'
        
    {% if is_incremental() %}
        -- sql code ...
    {% endif %}
```

![image](https://github.com/user-attachments/assets/10cdfd7c-a6c7-45cf-b099-390ebdce396f)

The workaround we found is to move a condition below the if block.

```
where
    date_hour_at::date <= expiry_date::date + 3
        
    {% if is_incremental() %}
        -- sql code ...
    {% endif %}

    and date_hour_at <= current_timestamp - interval '150 minutes'
```

![image](https://github.com/user-attachments/assets/d6b5bf71-04b7-42a2-837e-2f3fdb3fb77a)

**Comment by alanmcruickshank** [comment]
@francois-blanchard - there have been a bunch of changes to LT02 released very recently, which might help with this. As for your specific example I'm having trouble recreating it. If I use this query I get no errors:

```sql
select *
from my_table
where
    date_hour_at::date <= expiry_date::date + 3
    and date_hour_at <= current_timestamp - interval '150 minutes'

    {% if is_incremental() %}
        and this_other_condition is not null
    {% endif %}
```

Have you got a full query that I can use for testing?

**Comment by rhasenack** [comment]
Hey, I've been having the same problem. Just to add one example:

Here was my original code snippet: 

```
    where
        (last_active_at > created_at or user_id is not null)
        {% if  is_incremental() %}
            and created_at::date >= dateadd(day, -1, (select max(updated_at::date) - interval '1 day' as max_updated_at from {{ this }}))
        {% endif %}
```

This passes linting in my local environment, but in DBT CI it complains that the jinja lines should be indented only 4 spaces. 

This fixes it in CI but breaks it in my local environment:
```
    where
        (last_active_at > created_at or user_id is not null)
    {% if  is_incremental() %}
            and created_at::date >= dateadd(day, -1, (select max(updated_at::date) - interval '1 day' as max_updated_at from {{ this }}))
    {% endif %}
```

To make it worse, in DBT Cloud IDE it accepts the first version and also accepts this:

```
    where
        (last_active_at > created_at or user_id is not null)
    {% if  is_incremental() %}
            and created_at::date >= dateadd(day, -1, (select max(updated_at::date) - interval '1 day' from {{ this }}))
        {% endif %}
```

**Comment by pauln-relay** [comment]
Has anyone been able to find a workaround for this? We are experiencing this with LT02 in our staging CICD vs. local envs. I saw another post suggesting setting the dbt templater target to prod in the config so that the same models get looked up in dev and staging, but our staging environment is managed by a service account user with tight perms and network controls, so that isn't feasible.

For now, I'm just going to try to wrap the incremental blocks in ` --noqa: LT02` and hopefully that quiets it down, but I'd like to find a more permanent fix.

**Comment by soujh** [comment]
Hey everyone, 

Having the same issue here. 

Everything works find locally but fail in gitlab-ci.

**Comment by peterbud** [comment]
@nicor88 :  Could you check the solution provided in #5551 ? I think that might solve this issue as well.

**Comment by peterbud** [comment]
Closing due to lack of response
